NIFI-10625 Add support for HTTP/2 in Registry

exceptionfactory opened the pull request:
# Summary

[NIFI-10625](https://issues.apache.org/jira/browse/NIFI-10625) Adds optional support for HTTP/2 in NiFi Registry.

The implementation builds on the shared `nifi-jetty-configuration` capabilities, which already support HTTP/2. Additional changes include the introduction of a `nifi-security-ssl` for building an `SSLContext`, without the dependencies currently associated with `nifi-security-utils`.

The configuration includes a new property named `nifi.registry.web.https.application.protocols`, which defaults to `http/1.1` and supports enabling HTTP/2 by specifying `h2 http/1.1` for protocol negotiation.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [X] JDK 11
  - [X] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

thenatog commented:
Tested out the main NiFi Registry UI with the _nifi.registry.web.https.application.protocols_ property unset, and saw protocol version http/1.1 in Chrome. Added the property set to _nifi.registry.web.https.application.protocols=h2_ and restarted, and noted that the protocol was now h2 for the main UI and _/nifi-registry-docs/documentation_.

However, one thing I've noticed is that registry now seems to bind on all interfaces instead of just the one specified in configuration. For example, typically I have this set:
```
nifi.registry.web.https.host=mydomain.com
nifi.registry.web.https.port=18080
```
and in logs it will show:

`2022-10-17 17:33:43,038 INFO [main] o.apache.nifi.registry.jetty.JettyServer https://mydomain.com:18080/nifi-registry`

but with this change, potentially due to the Jetty changes, I'm now seeing:

```
2022-10-17 17:35:43,038 INFO [main] o.apache.nifi.registry.jetty.JettyServer NiFi Registry has started. The UI is available at the following URLs:
2022-10-17 17:35:43,038 INFO [main] o.apache.nifi.registry.jetty.JettyServer https://192.168.1.33:18080/nifi-registry
2022-10-17 17:35:43,038 INFO [main] o.apache.nifi.registry.jetty.JettyServer https://192.168.1.22:18080/nifi-registry
2022-10-17 17:35:43,038 INFO [main] o.apache.nifi.registry.jetty.JettyServer https://127.0.0.1:18080/nifi-registry
```

exceptionfactory commented:
Thanks for the review @thenatog!

Thanks for catching that problem with the host property configuration. The refactored Server Connector Factory was not setting the host property when defined. I updated the implementation and added unit tests to ensure that the property will be applied as expected.
